modules/services/nsi.js

```diff
@@ -0,0 +1,668 @@
+import { Matcher } from 'name-suggestion-index';
+
+import { fileFetcher, locationManager } from '../core';
+import { presetManager } from '../presets';
+
+// This service contains all the code related to the **name-suggestion-index** (aka NSI)
+// NSI contains the most correct tagging for many commonly mapped features.
+// See https://github.com/osmlab/name-suggestion-index  and  https://nsi.guide
+
+
+// DATA
+
+let _nsiStatus = 'loading';  // 'loading', 'ok', 'failed'
+let _nsi = {};
+
+// Sometimes we can upgrade a feature tagged like `building=yes` to a better tag.
+const buildingPreset = {
+  'building/commercial': true,
+  'building/government': true,
+  'building/hotel': true,
+  'building/retail': true,
+  'building/office': true,
+  'building/supermarket': true,
+  'building/yes': true
+};
+
+// Exceptions to the namelike regexes.
+// Usually a tag suffix contains a language code like `name:en`, `name:ru`
+// but we want to exclude things like `operator:type`, `name:etymology`, etc..
+const notNames = /:(colou?r|type|forward|backward|left|right|etymology|pronunciation|wikipedia)$/i;
+
+// Exceptions to the branchlike regexes
+const notBranches = /(coop|express|wireless|factory|outlet)/i;
+
+
+// PRIVATE FUNCTIONS
+
+// `setNsiSources()`
+// Adds the sources to iD's filemap so we can start downloading data.
+//
+function setNsiSources() {
+  const sources = {
+    'nsi_data': 'https://cdn.jsdelivr.net/npm/name-suggestion-index@6.0/dist/nsi.min.json',
```

Comment:
I don't think we'd want to automate it.. The CDN will already serve the latest file that matches the requested [semver](https://semver.org/) number, so 6.0.xxxxxx.

I've been following semver on the NSI side, so if there is a major bump from 5 to 6, that means there is some incompatibility in the code that need to be addressed.  I always put stuff like this in the [changelog](https://github.com/osmlab/name-suggestion-index/blob/main/CHANGELOG.md).

Basically my thinking is:
- major bump - maybe some code needs to be rewritten, maybe data file format is different
- minor bump - code should still work, file format same maybe with new properties (NSI might get a new tree)
- patch bump - code is same, file format is same, some data is new. (NSI publishes a new patch about every week).

Reply:
Got it. I agree.
Ok then I have only this small observation: When new major/minor packages version becomes available, maintainer would have to bump version in both package.json and url links. It would be more secure to have one singular place with version and for that info to propagate automatically wherever its needed.

Reply:
Oh yeah! I didn't understand before what you meant by automatization originally but I do now.  I think I can make a small improvement here so that this code constructs the URL to have the appropriate version number that matches whatever `package.json` has.

Reply:
I did this, check out 8db1c1f7f

Reply:
Can use this trick for a handful of other data files too, like osm-community-index, or the presets